testpb/1.pulsar.go

```diff
@@ -976,7 +1007,1122 @@ func (x *fastReflection_A) IsValid() bool {
 // "google.golang.org/protobuf/runtime/protoiface".Methods.
 // Consult the protoiface package documentation for details.
 func (x *fastReflection_A) ProtoMethods() *protoiface.Methods {
-	return nil
+	size := func(input protoiface.SizeInput) protoiface.SizeOutput {
+		x := input.Message.Interface().(*A)
+		if x == nil {
+			return protoiface.SizeOutput{
+				NoUnkeyedLiterals: struct{}{},
+				Size:              0,
+			}
+		}
+		var n int
+		var l int
+		_ = l
+		if x.Enum != 0 {
+			n += 1 + runtime.Sov(uint64(x.Enum))
+		}
+		if x.SomeBoolean {
+			n += 2
+		}
+		if x.INT32 != 0 {
+			n += 1 + runtime.Sov(uint64(x.INT32))
+		}
+		if x.SINT32 != 0 {
+			n += 1 + runtime.Soz(uint64(x.SINT32))
+		}
+		if x.UINT32 != 0 {
+			n += 1 + runtime.Sov(uint64(x.UINT32))
+		}
+		if x.INT64 != 0 {
+			n += 1 + runtime.Sov(uint64(x.INT64))
+		}
+		if x.SING64 != 0 {
+			n += 1 + runtime.Soz(uint64(x.SING64))
+		}
+		if x.UINT64 != 0 {
+			n += 1 + runtime.Sov(uint64(x.UINT64))
+		}
+		if x.SFIXED32 != 0 {
+			n += 5
+		}
+		if x.FIXED32 != 0 {
+			n += 5
+		}
+		if x.FLOAT != 0 || math.Signbit(float64(x.FLOAT)) {
+			n += 5
+		}
+		if x.SFIXED64 != 0 {
+			n += 9
+		}
+		if x.FIXED64 != 0 {
+			n += 9
+		}
+		if x.DOUBLE != 0 || math.Signbit(x.DOUBLE) {
+			n += 9
+		}
+		l = len(x.STRING)
+		if l > 0 {
+			n += 1 + l + runtime.Sov(uint64(l))
+		}
+		l = len(x.BYTES)
+		if l > 0 {
+			n += 2 + l + runtime.Sov(uint64(l))
+		}
+		if x.MESSAGE != nil {
+			l = runtime.MarshalFlagsToOptions(input.Flags).Size(x.MESSAGE)
```

Comment:
`runtime.MarshalFlagsToOptions` is used everywhere, let's instantiate the object at the top of the function so we avoid the unnecessary overhead each time we need it.